src/rebar.app.src

```diff
@@ -8,6 +8,7 @@
   {registered, []},
   {applications, [kernel,
                   stdlib,
+                  %% note: will be filtered out by rebar_prv_escriptize on Erlang/OTP 24+
                   hipe,
```

Comment:
since we got dialyzer required, we can do the filtering with the scripting mode for the app file. Something like this:

```
{application, rebar,
 [{description, "Rebar: Erlang Build Tool"},
  {vsn, "git"},
  {modules, []},
  {registered, []},
  {applications, [kernel,
                  stdlib,
                  sasl,
                  compiler,
                  crypto,
                  syntax_tools,
                  ...
                  inets,
                  eunit_formatters
                 | [hipe || _ <- [application:load(dialyzer)],
                            {ok, Apps} <- [application:get_key(dialyzer, applications)],
                            lists:member(hipe, Apps)] 
                ]},
  {env, [
      ...
        ]}
 ]}.
```

This would send the whole check to runtime and avoid special escriptize code.

Reply:
Oh to be clear we'd need a rebar.app.script file for this I believe, so a rename is in order